08-rsf-prep.qmd

```diff
@@ -90,104 +83,122 @@ mapview(bou_pts, color = "red", cex= 3)
 
 ```
 
+We can now combine our caribou locations and "background" locations into a single data set. We will retain the spatial information to allow us to easily extract the values in the corresponding raster stack
 
-We can now combine our caribou locations and "available" locations into a single dataset. We will retain the spatial information to allow us to easily extract the values in the corresponding raster stack 
+```{r, eval = FALSE}
 
+head(avail_points)
 
-```{r, eval = FALSE}
+head(bou_pts)
 
-avail_points 
+allpts <- bind_rows(bou_pts, avail_points) 
 
+head(allpts)
+# hmmmmm...........that looks weird what happened? 
 
-bou_pts
 
 # note we have slightly different column headers "geom" vs "geometry" 
 st_geometry(avail_points) = "geom" 
 
-
-allpts <- bind_rows(bou_pts, avail_points ) 
+allpts <- bind_rows(bou_pts, avail_points) 
 
 ```
 
+## Extract point values
 
-Next, we can read in our prepared raster stack as an .rds object. We can now use the **extract** function from the **terra** package to extract information for all layers in the raster stack for each of our points.  
-
-
-```{r, eval = FALSE}
+Next, we can read in our prepared raster stack as an .rds object. We can now use the **extract** function from the **terra** package to extract information for all layers in the raster stack for each of our points.
 
+```{r}
 # read in the raster stack 
-
-rstack <- readRDS(file.path("clean_data", "covars.RDS"))
-
+rstack <- readRDS("clean_data/covars.RDS")
 
 # extract all values in the raster stack for each location in the bou_pts file. 
-
 atts <- terra::extract(rstack, allpts)
 
+head(atts)
 
 # remove unused columns 
 # Could show how to use st_write() and explain the cbind(st_coordinates()) bit
-bou_full_pts <- cbind(allpts, atts) %>%
+bou_full_pts <- cbind(allpts, atts) |>
   select(-ID)
 
+```
 
-# export this as spatial file # or 
+Export as geo package or as csv table.
 
+```{r}
 
-bou_table <- bou_full_pts %>%
-  cbind(st_coordinates(.)) %>%
-  st_drop_geometry()
+# spatial file 
 
+write.csv(bou_full_pts, "clean_data/allpts_att.gpkg")
 
-# write out as csv 
 
-write.csv(bou_table, file.path("clean_data", "allpts_att.csv"))
+# write out as csv, keeping the XY values
 
+bou_table <- bou_full_pts %>%
+  cbind(st_coordinates(.)) |>
```

Comment:
The placeholder for the base pipe is `_` instead of `.` - did you try that? I know it doesn't work in all the same places as teh magrittr `%>%`/`.`?

Otherwise I think maybe 

```r
bou_table <- st_coordinates(bou_full_pts) |> 
  cbind(bou_full_pts) |> 
  st_drop_geometry()
```
 or something?